pkg/api/dashboard.go

```diff
@@ -143,6 +143,7 @@ func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
 		Url:                    dash.GetUrl(),
 		FolderTitle:            "General",
 		AnnotationsPermissions: annotationPermissions,
+		IsPublic:               dash.IsPublic,
```

Comment:
don't we need a sort of guid as well so we can share it publicly and later revoke / regenerate a new key?

Reply:
We should also consider how we can share with specific people / emails. 

Ddog has some interesting abilities here: 

* Share by public link / url
* Share to list of emails (each email get's their own link, which you can revoke if you later remove the email from the dashboard)
* Share to a domain (you get a link that requires each user to specify their email and ddog email a link to that email), this allow you to share a dashboard to a email domain without first specifying every email of that domain.

Reply:
@torkelo Thanks for the additional use cases.

This PR is just for persisting an absolutely basic config. We're anticipating the sharing config will become significantly more complicated in the future. Once we have a draft of the backend query engine, we'll make a separate pass for url routing and permissions.

Reply:
@ryantxu what's your thoughts on this? Let's say we want to make any dashboard public, and beyond the flag there are

* a public URL / key that can be revoked re-generated
* a list of emails, for who can access it (They get sent out unique access keys)
* share to a email domain, when some wants to access it they need to specify their email and we send them a access key

Given the future document storage, where would we store this additional public dashboard & access keys? Inside the dashboard document? that would not be possible as we cannot share the access keys. So would need to be more secure.

Reply:
@torkelo @ryantxu This toggle is just a high level enable/disable for a dashboard. Configuration will be stored in another table which will enable us to revoke URLs as well as switch off the public dashboards for a given table https://github.com/grafana/grafana/pull/48470